tests/linearGaussian_snle_test.py

```diff
@@ -502,3 +502,108 @@ def test_api_nle_sampling_methods(
         posterior.train(max_num_iters=10)
 
     posterior.sample(sample_shape=(num_samples,), show_progress_bars=False)
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("num_dim", (1, 2))
+@pytest.mark.parametrize("prior_str", ("uniform", "gaussian"))
+@pytest.mark.parametrize("model_str", ("zuko_maf", "zuko_nsf"))
+def test_c2st_nle_unconstrained_space(
+    num_dim: int, prior_str: str, model_str: str, mcmc_params_accurate: dict
+):
+    """Test SNL on linear Gaussian in unconstrained space.
+
+    Args:
+        num_dim: parameter dimension of the gaussian model
+        prior_str: one of "gaussian" or "uniform"
+
+    """
+    num_samples = 500
+    num_simulations = 3000
+    trials_to_test = [1]
+
+    # likelihood_mean will be likelihood_shift+theta
+    likelihood_shift = -1.0 * ones(num_dim)
+    # Use increased cov to avoid too small posterior cov for many trials.
+    likelihood_cov = 0.8 * eye(num_dim)
+
+    if prior_str == "gaussian":
+        prior_mean = zeros(num_dim)
+        prior_cov = eye(num_dim)
+        prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
+    else:
+        prior = BoxUniform(-2.0 * ones(num_dim), 2.0 * ones(num_dim))
+
+    def simulator(theta):
+        return linear_gaussian(theta, likelihood_shift, likelihood_cov)
+
+    # Use likelihood_nn with z_score_theta="transform_to_unconstrained"
+    density_estimator = likelihood_nn(
+        model_str,
+        hidden_features=60,
+        num_transforms=3,
+        z_score_theta="transform_to_unconstrained",
+        x_dist=prior,
```

Comment:
why do we pass the prior here? this should be a distribution in `x`, no? for a MVN prior it does not matter because both theta and x are unbounded, but for the box prior this is incorrect because x can be outside of the box when sampled with a mean close to the box boundary.